diagnostics-program/main.py

```diff
@@ -27,269 +26,131 @@
+        try:
+            diagnostics[key] = utils.config_search_param(command, param)
+        except Exception as e:
+            logging.error(e)
 
 
-def get_miner_diagnostics():
-    # Get miner diagnostics
-    # return MC - MD - MH - MN list
-    param_list = []
-    try:
-        miner_bus = dbus.SystemBus()
-        miner_object = miner_bus.get_object('com.helium.Miner', '/')
-        miner_interface = dbus.Interface(miner_object, 'com.helium.Miner')
-        try:
-            p2pstatus = miner_interface.P2PStatus()
-            param_list = [
-                str(p2pstatus[0][1]),
-                str(p2pstatus[1][1]),
-                str(p2pstatus[3][1]),
-                str(p2pstatus[2][1])
-            ]
-        except dbus.exceptions.DBusException:
-            param_list = [
-                "no",
-                "",
-                "0",
-                ""
-            ]
-    except Exception:
-        param_list = [
-            "no",
-            "",
-            "0",
-            ""
-        ]
-
-    return param_list
-
-
-def config_search_param(command, param):
-    """
-    input:
-        command: Command to execute
-        param: The parameter we are looking for in the response
-    return: True is exist, or False if doesn't exist
-    Possible exceptions:
-        TypeError: If the arguments passed to the function are not strings.
-    """
-    if type(command) is not str:
-        raise TypeError("The command must be a string value")
-    if type(param) is not str:
-        raise TypeError("The param must be a string value")
-    result = subprocess.Popen(command.split())
-    if param in result:
-        return True
+def write_public_keys_to_diag(data, diagnostics):
+    if data is not None and len(data) == 3:
+        keys = ["PK", "OK", "AN"]
+        for (param, key) in zip(data, keys):
+            diagnostics[key] = param
     else:
-        return False
-
-
-def writing_data(path, data):
-    """
-    input:
-        path - path to the file
-        data - data to write to file
-    Possible exceptions:
-    TypeError - if the path is not str.
-    FileNotFoundError - Directory does not exist in the path
-    PermissionError - No file permissions
-    """
-    if type(path) is not str:
-        raise TypeError("The path must be a string value")
-    try:
-        with open(path, 'w') as file:
-            file.write(data)
-    except FileNotFoundError:
-        raise FileNotFoundError(
-            f"Directory does not exist in the path: {path}"
+        logging.error(
+            "The public keys from the file were obtained with an unknown error"
         )
-    except PermissionError as e:
-        raise e
+
+
+def set_param_miner_diag(diagnostics):
+    param_miner_diag = utils.get_miner_diagnostics()
+    keys = ['MC', 'MD', 'MH', 'MN']
+    for (param, key) in zip(param_miner_diag, keys):
+        diagnostics[key] = param
+
+
+def write_info_to_files(prod_diagnostics, diagnostics):
```

Comment:
While this is very compact and efficiently written, it's not super readable in my opinion. The *order* of the mapping between `path_list` and `data_list` is critical here. If the order in any of these were to change, it would break things. Maybe a dict (`{'data_json': '/opt/nebraDiagnostics/html/diagnostics.json'}`) would make this more readable. Moreover, there is no test if this order were to get shuffled.